fix only allow these keys for signature

jitsealgra opened the pull request:
only allow the order_id and status key for the signature validation as given in the doc

[https://developer.rabobank.nl/omnikassa-payment-api](https://developer.rabobank.nl/omnikassa-payment-api)

jitsealgra committed:
fix only allow these keys for signature

NinkevdVeen commented on src/Message/Response/AbstractRabobankResponse.php:
```diff
@@ -58,10 +58,9 @@ protected function validateSignature()
-        unset($signatureData['timestamp']);
+
+        $allowedKeys = array_flip(['order_id', 'status']);
+        $signatureData = array_intersect_key($this->data, $allowedKeys);
```
see also: https://github.com/thephpleague/omnipay-rabobank/pull/14
As I understand it, this function is used for request, response and status. The aparams for these are different though, that's why they used this particular method before. I quote a dev: "To make it more future-proof, we need some way to let the response object tell us what fields to use to check the signature."

so in short, are we sure this will work in all three situations?

jitsealgra replied:
Well this will be a better option than what it currently is, if you want it for a more future-proof then this might be a way to go. Since Rabobank wil not change the signature as it currently is, then they must bring a new version if its deprecated. (except if it is an security issue)

jitsealgra replied:
Fixed per response can now have there own allowedKeys. Also for the payment_brands call that returns a call will get validated, so when there is a redirectUrl it will skip and do the old signatureValidation

jitsealgra committed:
fix only allow these keys for signature

NinkevdVeen commented:
I approve based on our conversation and the fact we probably can't do this locally and need it to go live first.

NinkevdVeen committed:
fix-testmode

fixed testmode by removing gateway call in between

NinkevdVeen committed:
isPaid

added isPaid, since it was missing. official docs say:
Rabo Omnikassa maakt onderscheid tussen de volgende eindstatussen van een order:
· “COMPLETED”: de order is succesvol betaald door de consument;
· “CANCELLED”: de order werd niet betaald omdat consument de betaling geannuleerd heeft;
· “EXPIRED”: de order werd niet betaald omdat de consument de betaling niet binnen de toegestane tijd heeft afgerond.
